src/ui-layout.js

```diff
@@ -796,6 +807,37 @@ angular.module('ui.layout', [])
             if(!element.hasClass('stretch')) element.addClass('stretch');
             if(!element.hasClass('ui-layout-container')) element.addClass('ui-layout-container');
 
+            scope.$watch('collapsed', function (val, old) {
+              if (old !== undefined && val !== old) {
```

Comment:
Actually during initialization, the equality can actually happen, which would in turn toggle the container against expectations.